rfcs/0014-vm-cycle-limits/0014-vm-cycle-limits.md

```diff
@@ -0,0 +1,111 @@
+
+## Cycle Measures
+
+Here we will specify the cycles needed by each CKB VM instructions or syscalls. Note right now in the RFC, we define hard rules for each instruction or syscall here, in future this might be moved into consensus rules so we can change them more easily.
+
+### Instruction Cycles
+
+All CKB VM instructions consume 1 cycle except the following ones:
+
+| Instruction | Cycles |
+|-------------|--------|
+| JALR        | 3      |
+| JAL         | 3      |
+| J           | 3      |
+| JR          | 3      |
+| BEQ         | 3      |
+| BNE         | 3      |
+| BLT         | 3      |
+| BGE         | 3      |
+| BLTU        | 3      |
+| BGEU        | 3      |
+| BEQZ        | 3      |
+| BNEZ        | 3      |
+| LD          | 2      |
+| SD          | 2      |
+| LDSP        | 2      |
+| SDSP        | 2      |
+| LW          | 3      |
+| LH          | 3      |
+| LB          | 3      |
+| LWU         | 3      |
+| LHU         | 3      |
+| LBU         | 3      |
+| SW          | 3      |
+| SH          | 3      |
+| SB          | 3      |
+| LWSP        | 3      |
+| SWSP        | 3      |
+| MUL         | 5      |
+| MULW        | 5      |
+| MULH        | 5      |
+| MULHU       | 5      |
+| MULHSU      | 5      |
+| DIV         | 16     |
+| DIVW        | 16     |
+| DIVU        | 16     |
+| DIVUW       | 16     |
+| REM         | 16     |
+| REMW        | 16     |
+| REMU        | 16     |
+| REMUW       | 16     |
+| ECALL       | 0      |
+| EBREAK      | 0      |
+
+### Syscall Cycles
+
+Each syscall in CKB has different rules for consuming cycles:
+
+#### Load TX
+
+*Load TX* syscall first consumes 10 initial cycles, it then measures the size of the serialized transaction data: for every single byte in the data, it consumes 10 more cycles.
+
+Note that even though the script only requires part of the serialized TX data, the syscall still charges based on the full serialized data size.
+
+#### Load Cell
+
+*Load Cell* syscall first consumes 100 initial cycles, it then measures the size of the serialized cell structure data: for every single byte in the serialized data, it consumes 100 more cycles.
+
+Notice the charged cycles here is 10 times the cycles charged in `Load Cell By Field` syscall, this is because we are discouraging the use of this syscall. One should only use this if they really need the full serialized Cell structure.
+
+Note that even though the script only requires part of the serialized Cell structure data, the syscall still charges based on the full serialized data size.
+
+#### Load Cell By Field
+
+*Load Cell By Field* syscall first consumes 10 initial cycles, it then measures the size of the serialized data from the specified field: for every single byte in the serialized data, it consumes 10 more cycles.
+
+Note that even though the script only requires part of the serialized data, the syscall still charges based on the full serialized data size.
```

Comment:
So the syscall charges based on serialized cell data size, instead of serialized field data size?

Reply:
There's a little confusion, so this is indeed charged based on serialized field data size.